src/components/Login/WelcomeMessageModalBody.tsx

```diff
@@ -0,0 +1,108 @@
+/* eslint-disable i18next/no-literal-string */
+import { useSWRConfig } from 'swr';
+
+import MoonIllustrationSVG from '../../../public/images/moon-illustration.svg';
+import Button from '../dls/Button/Button';
+import Carousel from '../dls/Carousel/Carousel';
+import Link, { LinkVariant } from '../dls/Link/Link';
+
+import styles from './WelcomeMessageModalBody.module.scss';
+
+import { completeOnboarding } from 'src/utils/auth/api';
+import { makeUserProfileUrl } from 'src/utils/auth/apiPaths';
+import UserProfile from 'types/auth/UserProfile';
+
+const WelcomeMessageModalBody = () => {
+  const { mutate, cache } = useSWRConfig();
+  const handleFinishOnboarding = async () => {
+    await completeOnboarding();
+    const userProfileData = cache.get(makeUserProfileUrl());
+    const newUserProfileData: UserProfile = {
+      ...userProfileData,
+      announcement: [], // clean up the announcement for this session
+    };
+    mutate(makeUserProfileUrl(), newUserProfileData);
```

Comment:
I believe this can be simplified like this:

```
mutate(makeUserProfileUrl(), (currentProfileData: UserProfile) => {
      return {
        ...currentProfileData,
        announcement: [], // clean up the announcement for this session
      };
    });
```

Reply:
nice 👍